### PR TITLE
Spacing

### PR DESCRIPTION
**Pull request by marcianx**

### CODE REVIEW

File: src/domTree.js [hunk 1]
```diff
@@ -15,17 +15,10 @@ import type Options from "./Options";
...
 const createClass = function(classes: string[]): string {
-    classes = classes.slice();
-    for (let i = classes.length - 1; i >= 0; i--) {
-        if (!classes[i]) {
-            classes.splice(i, 1);
-        }
-    }
-
-    return classes.join(" ");
+    return classes.filter(cls => cls).join(" ");
```
Comment (kevinbarabash) [inline src/domTree.js]: Nice!

File: src/buildCommon.js [hunk 2]
```diff
@@ -222,10 +222,10 @@ const boldsymbol = function(
 /**
  * Makes either a mathord or textord in the correct font and color.
  */
-const makeOrd = function<NODETYPE: "textord" | "mathord">(
+const makeOrd = function<NODETYPE: "spacing" | "textord" | "mathord">(
     group: ParseNode<NODETYPE>,
     options: Options,
-    type: NODETYPE,
+    type: NODETYPE | "textord",
```
Comment (kevinbarabash) [inline src/buildCommon.js]: Isn't `"textord"` already included in `NODETYPE`?

Comment (marcianx) [reply src/buildCommon.js]: You're right, that's confusing. I changed `type` to `"mathord" | "textord"` instead. Previous PR dealt with the case where `ParseNode<"mathord">` was called type `"mathord"` and similarly for `"textord"` so the original type made sense. Now, I've got `ParseNode<"spacing">` is called with type `"textord"` which means that I have to break the tight-knit relationship between the `ParseNode` type and `type`.

File: src/domTree.js (hunk 1 above)
Comment (ylemkimon) [reply src/domTree.js]: I'm not sure why, but `.filter(Boolean)` seems to be more commonly used.

Comment (marcianx) [reply src/domTree.js]: Fascinating. Never seen that myself. Would you prefer it in this case?

Comment (kevinbarabash) [reply src/domTree.js]: This is fine as is.

**Pull request merged**